data/static/codefixes/dbSchemaChallenge_2_correct.ts

```diff
@@ -1,19 +1,25 @@
-module.exports = function searchProducts () {
+module.exports = function searchProducts() {
   return (req: Request, res: Response, next: NextFunction) => {
-    let criteria: any = req.query.q === 'undefined' ? '' : req.query.q ?? ''
-    criteria = (criteria.length <= 200) ? criteria : criteria.substring(0, 200)
-    models.sequelize.query(
-        `SELECT * FROM Products WHERE ((name LIKE '%:criteria%' OR description LIKE '%:criteria%') AND deletedAt IS NULL) ORDER BY name`,
-        { replacements: { criteria } }
-      ).then(([products]: any) => {
-        const dataString = JSON.stringify(products)
+    let criteria: any = req.query.q === "undefined" ? "" : req.query.q ?? "";
+    criteria = criteria.length <= 200 ? criteria : criteria.substring(0, 200);
+    models.sequelize
+      .query(
+        "SELECT * FROM Products WHERE ((name LIKE '%:" +
+          criteria +
+          "%' OR description LIKE '%:" +
+          criteria +
+          "%') AND deletedAt IS NULL) ORDER BY name"
+      )
```

Comment:
🚫 **[Bearer]** <[javascript_lang_sql_injection](https://docs.bearer.com/reference/rules/javascript_lang_sql_injection)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>
# SQL injection vulnerability detected.
## Description
Including unsanitized data, such as user input or request data, in raw SQL queries makes your application vulnerable to SQL injection attacks.

## Remediations

❌ Avoid raw queries, especially those that contain unsanitized user input

```javascript
  var sqlite = new Sequelize("sqlite::memory:");
  sqlite.query("SELECT * FROM users WHERE ID = " + req.params.userId);
```

Instead, consider the following approaches when writing SQL queries

✅ Validate query input wherever possible

```javascript
  var rawId = req.params.userId
  if !(/[0-9]+/.test(rawId)) {
    // input is unexpected; don't make the query
  }
```

✅ Use prepared (or parameterized) statements when querying

Sequelize example -
```javascript
  var sqlite = new Sequelize("sqlite::memory:");
  sqlite.query(
    "SELECT * FROM users WHERE ID = ?",
    { replacements: [req.params.userId] },
    type: sequelize.QueryTypes.SELECT
  )
```

## Resources
- [OWASP SQL injection explained](https://owasp.org/www-community/attacks/SQL_Injection)
- [OWASP SQL injection prevention cheat sheet](https://cheatsheetseries.owasp.org/cheatsheets/SQL_Injection_Prevention_Cheat_Sheet.html)